src/formats/onenote.ts

```diff
@@ -0,0 +1,489 @@
+			try {
+				const tokenResponse = await requestUrl({
+					method: 'POST',
+					url: 'https://login.microsoftonline.com/common/oauth2/v2.0/token',
+					contentType: 'application/x-www-form-urlencoded',
+					body: requestBody.toString(),
+				});
+				const tokenData: tokenResponse = tokenResponse.json;
+				await this.signIn(tokenData.access_token);
+				clearInterval(intervalId);
+			}
+			catch (e) {
+				console.log(e);
+			}
+		}, deviceCodeRequest.interval * 1000); // Convert seconds into miliseconds
+	}
+	
+	async signIn(accessToken: string) {
+		this.requestParams = {
+			headers: {
+				Authorization: `Bearer ${accessToken}`,
+			},
+		};
+		await this.showSectionPickerUI();
+	}
+
+	async showSectionPickerUI() {
+		const response = await fetch(
+			'https://graph.microsoft.com/v1.0/me/onenote/notebooks?$expand=sections($select=id,displayName)&$select=id,displayName&$orderby=createdDateTime',
+			this.requestParams
+		);
+		const data = await response.json();
+		const notebooks: Notebook[] = data.value;
+
+		// Replace the sign in area to declutter the UI
+		this.contentArea.empty();
+
+		this.contentArea.createEl('h3', {
+			text: 'Choose what to import',
+			cls: 'modal-title',
+		});
+
+		notebooks.forEach((notebook) => {
+			this.contentArea.createEl('h5', {
+				text: notebook.displayName!,
+				cls: 'modal-title',
+			});
+
+			let sections: OnenoteSection[] | undefined | null = notebook?.sections;
+			let sectionGroups: SectionGroup[] | undefined | null = notebook?.sectionGroups;
+
+			if (sections) this.createSectionList(sections);
+
+			sectionGroups?.forEach((sectionGroup) => {
+				this.modal.contentEl.createEl('h6', {
+					text: sectionGroup.displayName!,
+				});
+				this.createSectionList(sectionGroup.sections!);
+			});
+		});
+	}
+
+	createSectionList(sections: OnenoteSection[]) {
+		const list = this.contentArea.createEl('ul');
+		sections?.forEach((section) => {
+			const listElement = list.createEl('li');
+
+			let label = listElement.createEl('label');
+
+			let checkbox = label.createEl('input');
+			checkbox.type = 'checkbox';
+
+			label.appendChild(document.createTextNode(section.displayName!));
+			label.createEl('br');
+
+			// Add/remove a section from this.selectedSections
+			checkbox.addEventListener('change', () => {
+				if (checkbox.checked) this.selectedSections.push(section);
+				else {
+					const index = this.selectedSections.findIndex((sec) => sec.id === section.id);
+					if (index !== -1) {
+						this.selectedSections.splice(index, 1);
+					}
+				}
+			});
+		});
+	}
+
+	async import(progress: ProgressReporter): Promise<void> {
+		let outputFolder = await this.getOutputFolder();
+		if (!outputFolder) {
+			new Notice('Please select a location to export to.');
+			return;
+		}
+
+		for (let section of this.selectedSections) {
+			let sectionFolder: TFolder = await this.createFolders(outputFolder.path + '/' + section.displayName);
+
+			let pages: OnenotePage[] = (await (await fetch(`
```

Comment:
It would be nice to have some error handling in here, so we can notify if we fail to load a page from a notebook.